hazelcast/src/main/java/com/hazelcast/spi/discovery/DiscoveryNode.java

```diff
@@ -40,8 +41,11 @@
  * {@link com.hazelcast.spi.discovery.NodeFilter} is configured, these
  * properties might be used for further refinement of the discovered nodes
  * based on whatever the filter decides.
+ *
+ * @since 3.6
  */
-public interface DiscoveredNode {
+@Beta
```

Comment:
For future maintainers it would be helpful to explain why an abstract class instead of an interface is being used. A lot of people see abstract classes as dirty and everything should go through interfaces.